tests/regressions/parallel/scan_different_inits.cpp

```diff
@@ -0,0 +1,166 @@
+    inclusive_scan(par, a.begin(), a.end(), c.begin(), 10,
+        [](int bar, int baz){ return bar*baz; });
+    exclusive_scan(par, a.begin(), a.end(), d.begin(), 100,
+        [](int bar, int baz){ return bar+baz; });
+    exclusive_scan(par, a.begin(), a.end(), e.begin(), 10,
+        [](int bar, int baz){ return bar*baz; });
+
+    detail::sequential_inclusive_scan(a.begin(), a.end(), b_ans.begin(), 100,
+        [](int bar, int baz){ return bar+baz; });
+    detail::sequential_inclusive_scan(a.begin(), a.end(), c_ans.begin(), 10,
+        [](int bar, int baz){ return bar*baz; });
+    detail::sequential_exclusive_scan(a.begin(), a.end(), d_ans.begin(), 100,
+        [](int bar, int baz){ return bar+baz; });
+    detail::sequential_exclusive_scan(a.begin(), a.end(), e_ans.begin(), 10,
+        [](int bar, int baz){ return bar*baz; });
+
+    HPX_TEST(std::equal(b.begin(), b.end(), b_ans.begin()));
+    HPX_TEST(std::equal(c.begin(), c.end(), c_ans.begin()));
+    HPX_TEST(std::equal(d.begin(), d.end(), d_ans.begin()));
+    HPX_TEST(std::equal(e.begin(), e.end(), e_ans.begin()));
+}
+
+void test_async_one(std::vector<int> a)
+{
+    using namespace hpx::parallel;
+    typedef typename std::vector<int>::iterator Iter;
+    typedef typename hpx::future<Iter> Fut_Iter;
+    std::size_t n = a.size();
+    std::vector<int> b(n), c(n), d(n), e(n);
+    std::vector<int> b_ans(n), c_ans(n), d_ans(n), e_ans(n);
+
+    Fut_Iter f_inc_add =
+        inclusive_scan(par(task), a.begin(), a.end(), b.begin(), 100,
+        [](int bar, int baz){ return bar+baz; });
+    Fut_Iter f_inc_mult =
+        inclusive_scan(par(task), a.begin(), a.end(), c.begin(), 10,
+        [](int bar, int baz){ return bar*baz; });
+    Fut_Iter f_exc_add =
+        exclusive_scan(par(task), a.begin(), a.end(), d.begin(), 100,
+        [](int bar, int baz){ return bar+baz; });
+    Fut_Iter f_exc_mult =
+        exclusive_scan(par(task), a.begin(), a.end(), e.begin(), 10,
+        [](int bar, int baz){ return bar*baz; });
+
+    detail::sequential_inclusive_scan(a.begin(), a.end(), b_ans.begin(), 100,
+        [](int bar, int baz){ return bar+baz; });
+    detail::sequential_inclusive_scan(a.begin(), a.end(), c_ans.begin(), 10,
+        [](int bar, int baz){ return bar*baz; });
+    detail::sequential_exclusive_scan(a.begin(), a.end(), d_ans.begin(), 100,
+        [](int bar, int baz){ return bar+baz; });
+    detail::sequential_exclusive_scan(a.begin(), a.end(), e_ans.begin(), 10,
+        [](int bar, int baz){ return bar*baz; });
+
+    f_inc_add.wait();
+    f_inc_mult.wait();
+    f_exc_add.wait();
+    f_exc_mult.wait();
+
+    HPX_TEST(std::equal(b.begin(), b.end(), b_ans.begin()));
+    HPX_TEST(std::equal(c.begin(), c.end(), c_ans.begin()));
+    HPX_TEST(std::equal(d.begin(), d.end(), d_ans.begin()));
+    HPX_TEST(std::equal(e.begin(), e.end(), e_ans.begin()));
+}
+
+int hpx_main(boost::program_options::variables_map& vm)
+{
+    std::vector<int> a1{1,1,2,2,3,3,4,4};
```

Comment:
Theoretically yes, but there are some issues with msvc and intel13. I think this simple case should be ok when used with a C++11 standard library implementation.